### PR TITLE
v2.4

### PR DESCRIPTION
**Pull request by ashpreetbedi**
- Remove LLM Task
- Add Async Run
- Add Messages Param
- Add Workflows

### CODE REVIEW

File: phi/assistant/assistant.py [hunk 1]
```diff
@@ -94,10 +107,12 @@ class Assistant(BaseModel):
...
-    # -*- Important: this setting determines if the input messages are formatted
-    # If True, phidata will add the system prompt, references, and chat history
-    # If False, the input messages are sent to the LLM as is
-    format_messages: bool = True
+    #
+    # -*- Assistant Messages
+    #
+    # -*- List of messages added to the messages list after the system prompt.
+    # Use these for few-shot learning or to provide additional context to the LLM.
+    add_messages: Optional[List[Union[Dict, Message]]] = None
```
Comment (sandangel) [inline phi/assistant/assistant.py:115]: This feels a bit confusing for me. Could be `additional_messages` instead? `add_messages` feel like a method for adding messages to a list.

File: phi/assistant/assistant.py [hunk 2]
```diff
@@ -218,62 +209,8 @@ def set_run_id(cls, v: Optional[str]) -> str:
     def streamable(self) -> bool:
         return self.output_model is None
```
Comment (sandangel) [inline phi/assistant/assistant.py:210]: Maybe with llama-index integration, we can stream partial intermediate pydantic object?

https://docs.llamaindex.ai/en/stable/examples/output_parsing/openai_pydantic_program/?h=stream+program#stream-partial-intermediate-pydantic-objects

**Comment by sandangel** [comment]
For `arun`, I feel it's repeating `run` code. Can we just code `arun`, then turn `run` into `asyncio.run(self.arun(...))`?

File: phi/assistant/assistant.py (hunk 1 above)
Comment (ashpreetbedi) [reply phi/assistant/assistant.py:115]: @sandangel agree, will update it to be `additional_messages`

File: phi/assistant/assistant.py (hunk 2 above)
Comment (ashpreetbedi) [reply phi/assistant/assistant.py:210]: @sandangel this is on the list for us to implement :) maybe i hack on it tonight :)

File: cookbook/assistants/add_messages.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from phi.assistant import Assistant
+from phi.llm.openai import OpenAIChat
+
+Assistant(
+    llm=OpenAIChat(model="gpt-3.5-turbo", stop="</answer>"),
+    system_prompt="What is the color of a banana? Provide your answer in the xml tag <answer>.",
+    add_messages=[{"role": "assistant", "content": "<answer>"}],
```
Comment (sandangel) [inline cookbook/assistants/add_messages.py:7]:
```suggestion
    additional_messages=[{"role": "assistant", "content": "<answer>"}],
```

Maybe a mypy type check would catch this bug.

**Comment by sandangel** [comment]
Is there a design document we can share for the `team` feature? It seems like we just renamed it to Workflow. I found that new concept is a bit confusing, from the crewai or autogen design POV.

File: cookbook/workflows/hackernews.py [hunk 4]
```diff
@@ -84,16 +57,24 @@ def get_user_details(username: str) -> str:
...
 hn_user_researcher = Assistant(
     name="HackerNews User Researcher",
-    role="Get information about hackernews users",
     tools=[get_user_details],
     show_tool_calls=True,
 )
 
-hn_assistant = Assistant(name="HackerNews Assistant", team=[hn_top_stories, hn_user_researcher], debug_mode=True)
-hn_assistant.print_response("Tell me about the users with the top 2 stores on hackernews?", markdown=True)
+hn_team = Workflow(
```
Comment (sandangel) [inline cookbook/workflows/hackernews.py:69]:
```suggestion
hn_workflow = Workflow(
```

File: phi/assistant/assistant.py (hunk 1 above)
Comment (ashpreetbedi) [reply phi/assistant/assistant.py]: @sandangel looking for a better name for this param :/

**Pull request merged**